src/isar/state_machine/state_machine.py

```diff
@@ -458,17 +496,17 @@ def publish_mission_status(self) -> None:
             retain=False,
         )
 
-    def publish_task_status(self) -> None:
-        """Publishes the current task status to the MQTT Broker"""
+    def publish_task_status(self, task: Task) -> None:
```

Comment:
Should task have a default value so that it can be an optional parameter?

Reply:
I don't think so because we will always be publishing task in relation to the executing task. This gives the opportunity to also publish for other tasks than the currently executing one.